app/controllers/templates-management/publish.js

```diff
@@ -19,39 +24,29 @@ export default class PublishController extends Controller {
 
   fetchPreview = task(async () => {
     this.currentVersion = '';
-    const publishedVersionContainer =
-      await this.model.container.publishedVersion.reload();
-    if (publishedVersionContainer) {
+    const currentPublishedTemplate = (
+      await this.store.query('template', {
+        filter: {
+          'derived-from': {
+            id: this.model.container.id,
+          },
+        },
+      })
+    )[0];
+    console.log(currentPublishedTemplate);
+    if (currentPublishedTemplate) {
       const publishedVersion =
-        await publishedVersionContainer.currentVersion.reload();
+        await currentPublishedTemplate.currentVersion.reload();
```

Comment:
Is it possible to do this in one go with the query for the template?

Reply:
Good point! I added an `include: 'current-version'` to the template query.